src/service/hidden-observer-impl.js

```diff
@@ -47,7 +47,8 @@ export class HiddenObserver {
     /** @const {!Document|!ShadowRoot} */
     this.root_ = opt_root || ampdoc.getRootNode();
 
-    const doc = opt_root ? opt_root.ownerDocument : this.root_;
+    const doc = opt_root ? opt_root.ownerDocument :
```

Comment:
Oh, it looks like opt_root can be of type `ShadowRoot`, so a similar fix should be applied here as well.

Reply:
Should be resolved with commit fcb55c6 . `doc` assignment in constructor now mimics the same routine in the constructor of `Url` in `src/service/url-impl.js`